File: tools/build_benchmark_jsons.py
#!/usr/bin/env python
# Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
import os
import sys
from util import build_path
from benchmark import read_json, write_json


def main(argv):
    if len(argv) == 2:
        build_dir = sys.argv[1]
    elif len(argv) == 1:
        build_dir = build_path()
    else:
        print "Usage: tools/build_benchmark_jsons.py [build_dir]"
        sys.exit(1)

    current_data_file = os.path.join(build_dir, "bench.json")
    all_data_file = "gh-pages/data.json"  # Includes all benchmark data.
    recent_data_file = "gh-pages/recent.json"  # Has 20 most recent results.

    assert os.path.exists(current_data_file)
    assert os.path.exists(all_data_file)

    new_data = read_json(current_data_file)
    all_data = read_json(all_data_file)
    all_data.append(new_data)

    write_json(all_data_file, all_data)
    write_json(recent_data_file, all_data[-20:])


if __name__ == '__main__':
    main(sys.argv)

File: tools/build_benchmark_jsons.js
// Copyright 2018-2026 the Deno authors. MIT license.
import { buildPath, existsSync, join } from "./util.js";

const currentDataFile = join(buildPath(), "bench.json");
const allDataFile = "gh-pages/data.json"; // Includes all benchmark data.
const allDataJsonlFile = "gh-pages/data.jsonl"; // JSONL version of all benchmark data.
const recentDataFile = "gh-pages/recent.json"; // Includes recent 20 benchmark data.
const recentDataJsonlFile = "gh-pages/recent.jsonl"; // JSONL version of recent benchmark data.

function readJson(filename) {
  return JSON.parse(Deno.readTextFileSync(filename));
}

function writeJson(filename, data) {
  return Deno.writeTextFileSync(filename, JSON.stringify(data));
}

function writeJsonl(filename, data) {
  return Deno.writeTextFileSync(
    filename,
    data.map((row) => JSON.stringify(row)).join("\n") + "\n",
  );
}

if (!existsSync(currentDataFile)) {
  throw new Error(`${currentDataFile} doesn't exist`);
}

if (!existsSync(allDataFile)) {
  throw new Error(`${allDataFile} doesn't exist`);
}

const newData = readJson(currentDataFile);
const allData = readJson(allDataFile);
allData.push(newData);
const allDataLen = allData.length;
const recentData = allData.slice(allDataLen - 20);

writeJson(allDataFile, allData);
writeJson(allDataJsonlFile, allData);
writeJson(recentDataFile, recentData);
writeJson(recentDataJsonlFile, recentData);
